### PR TITLE
fix: refactor pure checks smart contracts network admin

### LINKED ISSUE #12689: refactor pureChecks SmartContracts NetworkAdmin
Refactor pureChecks for Smart Contracts and Network Admin services. Move any checks which do not rely on state from preHandle and handle methods to pureChecks method for each transaction handler.

### PR DESCRIPTION
**Pull request by kimbor**
**Description**:
Cherry pick of PRs related to pureChecks in Smart Contract and Network Admin services.

Cherry pick the following from `release/0.49` to `develop`:
- #12748 
- #12874 
- #12878 

**Related issue(s)**:
Fixes issue #12689

### CODE REVIEW

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractSystemUndeleteHandler.java [hunk 1]
```diff
@@ -46,18 +46,20 @@ public ContractSystemUndeleteHandler() {
...
     public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
+        // this will never actually get called
+        // because pureChecks will always throw
         throw new PreCheckException(NOT_SUPPORTED);
     }
 
     @Override
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
-        // nothing to do
+        throw new PreCheckException(NOT_SUPPORTED);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractSystemUndeleteHandler.java:56]: Should we throw in `pureChecks` ? I am not sure why it was reverted in `UncheckedSubmit` https://github.com/hashgraph/hedera-services/pull/12874/files

Comment (kimbor) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractSystemUndeleteHandler.java:56]: Per discussion, this is the correct behavior.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @kimbor

**Review by lukelee-sl — COMMENTED** [review]
I don't have context on these changes.  It does seem to have the mentioned 0.49 changes cherry picked except for the change to `NetworkUncheckedSubmitHandler.java`.  Should that be included in the change set?

**Comment by kimbor** [comment]
> I don't have context on these changes. It does seem to have the mentioned 0.49 changes cherry picked except for the change to `NetworkUncheckedSubmitHandler.java`. Should that be included in the change set?

The `NetworkUncheckedSubmitHandler` changes were included in PR #12748 but then reverted in PR #12874. Since both of those `0.49` PRs are cherry-picked as part of this `develop` PR, the end result is no changes to `NetworkUncheckedSubmitHandler` in this change set.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM!

**Review by lukelee-sl — COMMENTED** [review]
LGTM.

**Review by lukelee-sl — APPROVED** [review]

**Pull request merged**